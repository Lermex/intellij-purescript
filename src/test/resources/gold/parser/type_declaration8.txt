Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('RandomConsoleEffects')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('eff')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('random')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Array')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('User')
                  PsiElement())(')')
              PsiElement(,)(',')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(Qualified)
                    PsiElement(proper name)('CONSOLE')
                    PsiElement(whitespace)(' ')
              PsiElement(|)('|')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(TypeVar)
                PSASTWrapperElement(TypeConstructor)
                  <empty list>
                PSASTWrapperElement(GenericIdentifier)
                  PsiElement(identifier)('eff')
                  PsiElement(whitespace)(' ')
            PsiElement())(')')
            PsiElement(whitespace)('\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('RandomConsoleEffects')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('eff')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PsiElement(whitespace)(' ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('random')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PsiElement(()('(')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Array')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('a')
                  PsiElement())(')')
              PsiElement(,)(',')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(LocalIdentifier)
                PSASTWrapperElement(identifier)
                  PsiElement(identifier)('console')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(Qualified)
                    PsiElement(proper name)('CONSOLE')
                    PsiElement(whitespace)(' ')
              PsiElement(|)('|')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(TypeVar)
                PSASTWrapperElement(TypeConstructor)
                  <empty list>
                PSASTWrapperElement(GenericIdentifier)
                  PsiElement(identifier)('eff')
                  PsiElement(whitespace)(' ')
            PsiElement())(')')
            PsiElement(whitespace)('\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('RandomConsoleEffect')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(GenericIdentifier)
              PsiElement(identifier)('a')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              <empty list>
            PsiElement())(')')
            PsiElement(whitespace)('\n')
      PSASTWrapperElement(TypeSynonymDeclaration)
        PsiElement(type)('type')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PsiElement(proper name)('RandomConsoleEffect')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('a')
          PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Type)
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('RandomConsoleEffects')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(TypeAtom)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('User')
            PsiElement())(')')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(TypeAtom)
            PsiElement(()('(')
            PSASTWrapperElement(Row)
              <empty list>
            PsiElement())(')')