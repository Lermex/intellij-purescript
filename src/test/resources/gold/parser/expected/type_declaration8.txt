Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('Main')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSTypeDeclarationImpl(TYPE_DECLARATION)
        PsiElement(PSTokenType.type)('type')
        PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
          PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
            PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffects')
        PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
          PSDeclarationRefIdImpl(DECLARATION_REF_ID)
            PsiElement(PSTokenType.ID)('eff')
        PsiElement(PSTokenType.=)('=')
        PsiElement(PSTokenType.()('(')
        PSRowItemListImpl(ROW_ITEM_LIST)
          PSRowItemImpl(ROW_ITEM)
            PsiElement(PSTokenType.ID)('random')
            PsiElement(PSTokenType.::)('::')
            PsiElement(PSTokenType.()('(')
            PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
              PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('Array')
              PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('User')
                PsiElement(PSTokenType.))(')')
          PsiElement(PSTokenType.,)(',')
          PSRowItemImpl(ROW_ITEM)
            PsiElement(PSTokenType.ID)('console')
            PsiElement(PSTokenType.::)('::')
            PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
              PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('CONSOLE')
          PsiElement(PSTokenType.|)('|')
          PsiElement(PSTokenType.ID)('eff')
        PsiElement(PSTokenType.))(')')
    PSDeclarationsImpl(DECLARATIONS)
      PSDeclarationImpl(DECLARATION)
        PSTypeDeclarationImpl(TYPE_DECLARATION)
          PsiElement(PSTokenType.type)('type')
          PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
            PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
              PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffects')
          PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
            PSDeclarationRefIdImpl(DECLARATION_REF_ID)
              PsiElement(PSTokenType.ID)('a')
            PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
              PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                PsiElement(PSTokenType.ID)('eff')
          PsiElement(PSTokenType.=)('=')
          PsiElement(PSTokenType.()('(')
          PSRowItemListImpl(ROW_ITEM_LIST)
            PSRowItemImpl(ROW_ITEM)
              PsiElement(PSTokenType.ID)('random')
              PsiElement(PSTokenType.::)('::')
              PsiElement(PSTokenType.()('(')
              PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('Array')
              PSParenthesisedDeclarationRefIdListImpl(PARENTHESISED_DECLARATION_REF_ID_LIST)
                PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                  PsiElement(PSTokenType.ID)('a')
                PsiElement(PSTokenType.))(')')
            PsiElement(PSTokenType.,)(',')
            PSRowItemImpl(ROW_ITEM)
              PsiElement(PSTokenType.ID)('console')
              PsiElement(PSTokenType.::)('::')
              PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('CONSOLE')
            PsiElement(PSTokenType.|)('|')
            PsiElement(PSTokenType.ID)('eff')
          PsiElement(PSTokenType.))(')')
      PSDeclarationsImpl(DECLARATIONS)
        PSDeclarationImpl(DECLARATION)
          PSTypeDeclarationImpl(TYPE_DECLARATION)
            PsiElement(PSTokenType.type)('type')
            PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
              PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffect')
            PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
              PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                PsiElement(PSTokenType.ID)('a')
            PsiElement(PSTokenType.=)('=')
            PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
              PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                PsiElement(PSTokenType.ID)('a')
            PsiElement(PSTokenType.()('(')
            PsiElement(PSTokenType.))(')')
        PSDeclarationsImpl(DECLARATIONS)
          PSDeclarationImpl(DECLARATION)
            PSTypeDeclarationImpl(TYPE_DECLARATION)
              PsiElement(PSTokenType.type)('type')
              PSDeclarationRefTypeListImpl(DECLARATION_REF_TYPE_LIST)
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffect')
              PSDeclarationRefIdListImpl(DECLARATION_REF_ID_LIST)
                PSDeclarationRefIdImpl(DECLARATION_REF_ID)
                  PsiElement(PSTokenType.ID)('a')
              PsiElement(PSTokenType.=)('=')
              PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                PsiElement(PSTokenType.()('(')
                PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('RandomConsoleEffects')
                PSParenthesisedDeclarationRefTypeListImpl(PARENTHESISED_DECLARATION_REF_TYPE_LIST)
                  PSDeclarationRefTypeImpl(DECLARATION_REF_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('User')
                  PsiElement(PSTokenType.))(')')
              PsiElement(PSTokenType.()('(')
              PsiElement(PSTokenType.))(')')